pytorch_lightning/trainer/training_loop.py

```diff
@@ -545,14 +564,13 @@ def run_training_batch(self, batch, batch_idx):
                 def optimizer_closure():
                     # forward pass
                     with self.profiler.profile('model_forward'):
-                        output = self.training_forward(
+                        output_dict = self.training_forward(
                             split_batch, batch_idx, opt_idx, self.hiddens)
 
-                    closure_loss = output[0]
-                    progress_bar_metrics = output[1]
-                    log_metrics = output[2]
-                    callback_metrics = output[3]
-                    self.hiddens = output[4]
+                        # format and reduce outputs accordingly
+                        processed_output = self.process_output(output_dict, train=True)
```

Comment:
This doesn't detach the members of the output_dict, so gradient trees will remain in memory

Reply:
You're absolutely right ! Thank you for pointing this out.
Have a look at the [latest commit](https://github.com/PyTorchLightning/pytorch-lightning/pull/1357/commits/39cecdba278bf8143bd746b934a7198d98d0237a) and tell me if this fix is OK with you.

Reply:
Awesome, that fixes the problem AFAICT, nice work! :)